### PR TITLE
Enhanced plugin handler tests

### PR DESCRIPTION
**Pull request by tpantelis**
Refactored the tests too reuse code and added more tests cases.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

### CODE REVIEW

File: plugin/lighthouse/handler.go [hunk 1]
```diff
@@ -14,38 +14,43 @@ import (
...
 
-	qname := state.QName()
 	// qname: mysvc.default.svc.example.org.
 	// zone:  example.org.
 	// Matches will return zone in all lower cases
 	zone := plugin.Zones(lh.Zones).Matches(qname)
 	if zone == "" {
-		return plugin.NextOrFailure(lh.Name(), lh.Next, ctx, w, r)
+		log.Debugf("Request does not match configured zones %v", lh.Zones)
+		return lh.nextOrFailure(state.Name(), ctx, w, r, dns.RcodeNotZone, "No matching zone found")
```
Comment (tpantelis) [inline plugin/lighthouse/handler.go]: @vthapar Note I changed this to call `lh.nextOrFailure` method to use the fall-through as in the other code paths below. Is this OK or was there a particular reason you called  `plugin.NextOrFailure` directly here?

Comment (vthapar) [reply plugin/lighthouse/handler.go]: Just missed this in refactoring.

File: plugin/lighthouse/handler_test.go [hunk 2]
```diff
@@ -160,90 +157,72 @@ func testNext() {
...
+			})
 		})
-		It("Should call next plugin", func() {
-			m := tc.Msg()
-			code, err := lh.ServeDNS(context.TODO(), rec, m)
-			Expect(err).To(HaveOccurred())
-			Expect(code).Should(Equal(tc.Rcode))
+	})
+
+	When("type A DNS query for a non-matching lighthouse zone and non-matching fallthrough zone", func() {
```
Comment (vthapar) [inline plugin/lighthouse/handler_test.go]: Do we have a test case when no explicit fallthrough zone is configured? It means "all zones match" and it should fallthrough. I think we had test case for it, just wanted to confirm.

Comment (tpantelis) [reply plugin/lighthouse/handler_test.go]: So I added a new test case with `lh.Fall = fall.F{}` but it doesn't fall through. Looking at the `Fall.Through` code, if `Zones` is empty then `Matches` returns `""` which fails the check `!= ""`. So there must be at least one matching zone.  Is this what we expect?

**Pull request merged**